feat(pattern): update German commercial registration number regex to allow underscores

typecastcloud opened the pull request:
## Description

Adjusted pattern for german crn with underscore separated court code and identifier.

## Why

Clearing house allows courts to use underscores in the commercial registration number.

## Issue

Refs: eclipse-tractusx/portal-frontend-registration#362
Depends on: eclipse-tractusx/portal-backend#1311

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/admin/Dev%20Process/How%20to%20contribute.md#commits-branches-and-pull-requests-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

evegufy commented on CHANGELOG.md:
```diff
@@ -6,7 +6,8 @@
 
-- updated validation of region code for tagus release [#1513](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/357)
+- updated validation of region code for tagus release [#357](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/357)
+- allow underscores in german crn [#364](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/364)
```
```suggestion
- allow underscores in german commercial registration number [#364](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/364)
```

evegufy reviewed (changes requested):
@typecastcloud [SanaSalem01](https://github.com/SanaSalem01) I think this change won't be sufficient to support the user through the registration flow. For instance, I tested such [a numeric pattern only](https://github.com/eclipse-tractusx/portal-frontend-registration/blob/098b8659f68f633f48883a9dcfed29fa5b138c83/src/types/testdata/crn.ts#L23-L25) in the last couple of days , and it was denied clearinghouse during the registration approval, so we shouldn't consider such a pattern as valid.
Please clarify with clearinghouse which patterns they consider valid.

typecastcloud commented:
> @typecastcloud [SanaSalem01](https://github.com/SanaSalem01) I think this change won't be sufficient to support the user through the registration flow. For instance, I tested such [a numeric pattern only](https://github.com/eclipse-tractusx/portal-frontend-registration/blob/098b8659f68f633f48883a9dcfed29fa5b138c83/src/types/testdata/crn.ts#L23-L25) in the last couple of days , and it was denied clearinghouse during the registration approval, so we shouldn't consider such a pattern as valid. Please clarify with clearinghouse which patterns they consider valid.

Hi @evegufy, many cases in the unit tests were only added to test boundaries of the pattern and are not real Ids. You are right that this is confusing and we should create a clearer list of real and invalid identifiers for both frontend and backend.

F1103R_HRB98814 was mentioned in this document https://github.com/eclipse-tractusx/portal-frontend-registration/issues/362#issuecomment-2697457613 and is an expected value from clearing house.

typecastcloud commented:
> > Hi @evegufy, many cases in the unit tests were only added to test boundaries of the pattern and are not real Ids. You are right that this is confusing and we should create a clearer list of real and invalid identifiers for both frontend and backend.
> > F1103R_HRB98814 was mentioned in this document [#362 (comment)](https://github.com/eclipse-tractusx/portal-frontend-registration/issues/362#issuecomment-2697457613) and is an expected value from clearing house.
> 
> Hi @typecastcloud it's good to update the list of eal and invalid identifiers for both frontend and backend. Still, the main point I want to transfer is that such only numbers pattern are still accepted by the application and that that should be changed. Therefore, my recommendation to clarify with clearinghouse which patterns they consider valid.
> 
> Please also see [#362 (comment)](https://github.com/eclipse-tractusx/portal-frontend-registration/issues/362#issuecomment-2703891740)

Hi @evegufy, I improved the Hint and created a BE PR eclipse-tractusx/portal-backend#1311. While there is probably a better way to guide users in a later iteration, this complexity also existed in ELBE and I don't see a clean way to improving on it without other risks in this version.

evegufy commented:
> > Hi @evegufy, many cases in the unit tests were only added to test boundaries of the pattern and are not real Ids. You are right that this is confusing and we should create a clearer list of real and invalid identifiers for both frontend and backend.
> > F1103R_HRB98814 was mentioned in this document [#362 (comment)](https://github.com/eclipse-tractusx/portal-frontend-registration/issues/362#issuecomment-2697457613) and is an expected value from clearing house.
> 
> Hi @typecastcloud it's good to update the list of eal and invalid identifiers for both frontend and backend. Still, the main point I want to transfer is that such only numbers pattern are still accepted by the application and that that should be changed. Therefore, my recommendation to clarify with clearinghouse which patterns they consider valid.
> 
> Please also see [#362 (comment)](https://github.com/eclipse-tractusx/portal-frontend-registration/issues/362#issuecomment-2703891740)

@typecastcloud very good that you improve that. are you going to address still [new input provided by Aleks from Clearinghouse](https://github.com/eclipse-tractusx/portal-frontend-registration/issues/362#issuecomment-2706251674) and the [issue with only numeric input](https://github.com/eclipse-tractusx/portal-frontend-registration/issues/362#issuecomment-2706591775)?

evegufy reviewed (changes requested):
https://github.com/eclipse-tractusx/portal-frontend-registration/pull/364#issuecomment-2706601341

typecastcloud commented:
@evegufy @manojava-gk 
German CRN always needs to start with a letter now. I think this will at least prevent numbers only identifiers.
New BE PR also includes this with new separation for countries: https://github.com/eclipse-tractusx/portal-backend/pull/1316

evegufy commented on src/locales/en/translations.json:
```diff
@@ -76,7 +76,7 @@
     "IN_VAT_ID": "Please enter a valid number. Hint: Alphanumeric with hyphen with 5 to 15 digits.",
     "IN_LEI_CODE": "Please enter a valid number. Hint: Alphanumeric with exact 20 digits.",
-    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid number. Hint: Alphanumeric with hyphen and space.",
+    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid identifier. Hint: Starting with court name or code and followed by registered identifier.",
```
As discussed yesterday, it was also recommended to add examples.
```suggestion
    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid identifier. Hint: Starting with court name or code and followed by registered identifier. Examples: HRB 98814, München HRB 175450, F1103R_HRB98814.",
```

typecastcloud replied:
```suggestion
    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid identifier. Hint: Starting with court name or code and followed by registered identifier. Examples: Mannheim HRB 98814, München HRB 175450, F1103R_HRB98814.",
```

typecastcloud commented on src/locales/de/translations.json:
```diff
@@ -76,7 +76,7 @@
     "IN_VAT_ID": "Please enter a valid number. Hint: Alphanumeric with hyphen with 5 to 15 digits.",
     "IN_LEI_CODE": "Please enter a valid number. Hint: Alphanumeric with exact 20 digits.",
-    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid number. Hint: Alphanumeric with hyphen and space.",
+    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid identifier. Hint: Starting with court name or code and followed by registered identifier.",
```
```suggestion
    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid identifier. Hint: Starting with court name or code and followed by registered identifier. Examples: Mannheim HRB 98814, München HRB 175450, F1103R_HRB98814.",
```

typecastcloud commented on src/locales/de/translations.json:
```diff
@@ -76,7 +76,7 @@
     "IN_VAT_ID": "Please enter a valid number. Hint: Alphanumeric with hyphen with 5 to 15 digits.",
     "IN_LEI_CODE": "Please enter a valid number. Hint: Alphanumeric with exact 20 digits.",
-    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid number. Hint: Alphanumeric with hyphen and space.",
+    "DE_COMMERCIAL_REG_NUMBER": "Please enter a valid identifier. Hint: Starting with court name or code and followed by registered identifier.",
```
We should probably not add the HRB example without court. It was advised against being used. I pushed alternative changes.

typecastcloud commented:
> Did you verify that the hints for the DE and the other countries are consistent with the changes introduced with [eclipse-tractusx/portal-backend#1316](https://github.com/eclipse-tractusx/portal-backend/pull/1316)?

I checked the difference and BE adjusted their changes to this PR. I couldn't find any discrepancy.

evegufy reviewed (commented):
please also update pr title/description and changelog entry to the current content of the pr.

typecastcloud commented on CHANGELOG.md:
```diff
@@ -6,7 +6,8 @@
 
-- updated validation of region code for tagus release [#1513](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/357)
+- updated validation of region code for tagus release [#357](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/357)
+- allow underscores in german commercial registration number [#364](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/364)
```
```suggestion
- allow underscores and prevent numbers only in german commercial registration number [#364](https://github.com/eclipse-tractusx/portal-frontend-registration/pull/364)
```

typecastcloud commented:
> I checked the difference and BE adjusted their changes to this PR. I couldn't find any discrepancy.

Not quite sure which part you are talking about. Backend error message? Or do you mean all the other hints/error in frontend?

evegufy commented:
> > I checked the difference and BE adjusted their changes to this PR. I couldn't find any discrepancy.
> 
> Not quite sure which part you are talking about. Backend error message? Or do you mean all the other hints/error in frontend?

I meant if the newly enabled identifiers in the backend for the different country code align with the hints and regex in the frontend, is that the case?

typecastcloud commented:
> > > I checked the difference and BE adjusted their changes to this PR. I couldn't find any discrepancy.
> > 
> > 
> > Not quite sure which part you are talking about. Backend error message? Or do you mean all the other hints/error in frontend?
> 
> I meant if the newly enabled identifiers in the backend for the different country code align with the hints and regex in the frontend, is that the case?

Yes, Backend is copy paste from frontend regex afaik. I don't like the quality of the hints in general (very generic without examples). They need proper UX refinement at some point.

evegufy commented:
> Yes, Backend is copy paste from frontend regex afaik. I don't like the quality of the hints in general (very generic without examples). They need proper UX refinement at some point.

Good that you checked that, I suggest to open an enhancement issue for the quality of the hints in general to be taken up for the next release.

typecastcloud commented:
Hi @evegufy @manojava-gk, we found that registration of french companies via commercial registration number is limited to SIREN identifier only. SIRET was never supported / tested by anyone. We added adjustment to both FE and BE in existing PRs and adjusted descriptions.

- new REGEX
- new Hint

CC: @tfjanjua @dorkabcox

typecastcloud commented:
> @typecastcloud just for the reminder, the DE translations are still in English.

That is the case for all identifier hints/errors. Out of scope imo.
